File: problem1/fixed/B01055680.py
def maxValidWindowSum(nums, k):
    windowSum = 0
    negativeCount = 0
    maxSum = float('-inf')


    for i in range(0, k):
        windowSum += nums[i]
        if nums[i] < 0:
            negativeCount += 1

            if negativeCount <= 1:
                maxSum = windowSum

                for i in range(k, len(nums)):
                    if nums[i - k] < 0:
                        negativeCount -= 1
                        windowSum -= nums[i - k]
                        windowSum += nums[i]
                        if nums[i] < 0:
                            negativeCount += 1
                            if negativeCount <= 1:
                                maxSum = max(maxSum, windowSum)


                                return (0 if maxSum == float('-inf') else maxSum)

File: problem1/java/B01055680.java
public class Solution {
public static int maxValidWindowSum(int[] nums, int k) {
    int windowSum = 0;
    int negativeCount = 0;
    int maxSum = Integer.MIN_VALUE;


    for (int i = 0; i < k; i++) {
        windowSum += nums[i];
        if (nums[i] < 0) {
            negativeCount++;
        }
    }

    if (negativeCount <= 1) {
        maxSum = windowSum;
    }

    for (int i = k; i < nums.length; i++) {
        if (nums[i - k] < 0) {
            negativeCount--;
        }
        windowSum -= nums[i - k];
        windowSum += nums[i];
        if (nums[i] < 0) {
            negativeCount++;
        }
        if (negativeCount <= 1) {
            maxSum = Math.max(maxSum, windowSum);
        }
    }


    return maxSum == Integer.MIN_VALUE ? 0 : maxSum;
}
}